code/modules/chemistry/tools/robospray.dm

```diff
@@ -69,19 +87,33 @@
 		"<span class='notice'>You inject [amt_prop] units of [propername]. [src] now contains [botreagents[currentreagent] - amt_prop] units.</span>")
 		logTheThing("combat", user, M, "uses a cybernetic hypospray to inject [constructTarget(M,"combat")] with [amt_prop] units of [propername] at [log_loc(user)].")
 
-		M.reagents.add_reagent(botreagents[picker], amt_prop)
+		M.reagents.add_reagent(currentreagent, amt_prop)
 		botreagents[currentreagent] = botreagents[currentreagent] - amt_prop
 		tooltip_rebuild = 1
+		update_icon()
 		playsound(get_turf(M), src.sound_inject, 80, 0)
 		return 0
 
 	process()
 		..()
+		var/refill_amount = (ticker.round_elapsed_ticks - last_refill) / 29
```

Comment:
i'd ideally want to see mult obtained from the base item process, so the /29 and stuff isn't necessary, and you have capped delays and such like the mob process mult

also it should be 2.9 SECONDS

Reply:
Should we also modify https://github.com/goonstation/goonstation/blob/master/code/datums/controllers/process/items.dm in this pull request?

Reply:
It would actually be here: https://github.com/goonstation/goonstation/blob/ef77f60edcc21430d6f1adc4d888afb3672ab2bd/code%2Fobj%2Fitem.dm#L771

Reply:
I was thinking it would pass the tick duration multiplier into that function as an argument. Or were you suggesting a different code change than that?

Reply:
the tick duration multiplier would be first on the base item process, and then pass-through inherited in the hypo's item process